web/index.html

```diff
@@ -552,64 +615,158 @@ <h3>Used in</h3>
+                    window.location.reload();
+                },
+                dragStart() {
+                    this.isDraggingBookmarklet = true;
+                },
+                dragEnd() {
+                    this.isDraggingBookmarklet = false;
+                },
+                preventClick(event) {
+                    this.isShaking = true;
+                    setTimeout(() => {
+                        this.isShaking = false;
+                    }, 250);
+                    event.preventDefault();
                 },
                 formatNumber: function (n) {
                     if (n === undefined || n <= 0) {
                         return "#####"
                     }
                     return n.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
                 },
+                elementTitle: function (element) {
+                    return this.saveDataExists ? (this.saveData.has(element) ? 'Crafted' : 'Not crafted') : '';
+                },
+                getChunk: function (value) {
+                    let base64 = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-='
+                    let result = 0
+                    for (let i = 0; i < value.length; i++) {
+                        result = (result * 64 + base64.indexOf(value[i]))
+                    }
+                    return Math.floor(result / 100)
+                },
                 loadChunk: async function (chunk) {
-                    return fetch(DATA_PREFIX + `chunks/chunk-${chunk}.json`)
-                        .then(response => {
-                            console.log("Loaded chunk", chunk);
-                            return response.json()
-                        })
-                        .then(data => {
-                            let [sep1, sep2, sep3] = [":3", ":2", ":1"]
-                            chunkData = Object.fromEntries(Object.entries(data).map(x => {
-                                let [from, to, hiddenFrom, hiddenTo] = x[1].split(sep1)
-                                if (from == "") {
-                                    from = []
-                                } else {
-                                    from = from.split(sep2).map(x => x.split(sep3))
-                                }
-                                if (to == "") {
-                                    to = []
-                                } else {
-                                    to = to.split(sep2).map(x => x.split(sep3))
-                                }
-                                if (hiddenFrom == "") {
-                                    hiddenFrom = []
-                                } else {
-                                    hiddenFrom = hiddenFrom.split(sep2).map(x => x.split(sep3))
-                                }
-                                if (hiddenTo == "") {
-                                    hiddenTo = []
+                    if (this.chunkLoadState[chunk]) {
```

Comment:
Load chunk is pretty straightforward, we first check if the current chunk is being loaded or has already been loaded, if so return the previous result. The return value is of type Promise, which can be awaited on nicely.
Next we setup a worker using the CHUNK_WORKER_PATH, which is just loadChunkWorker.js.
Then we setup some handlers for when it is done or if it hits an error. And then we start it using a post message by sending it the data it should use to run.